File: ios/atb/AtBRootView.m
#import "AtBRootView.h"
#import <React/RCTRootView.h>

@implementation AtBRootView

- (void) traitCollectionDidChange: (UITraitCollection *) previousTraitCollection
{
  [super traitCollectionDidChange: previousTraitCollection];
  
  if (@available(iOS 13.0, *)) {
    if (self.traitCollection.userInterfaceStyle != previousTraitCollection.userInterfaceStyle) {
      [self setBackgroundByTrait];
    }
  }
}

- (void)setBackgroundByTrait
{
  if (@available(iOS 13.0, *)) {
    if ([UITraitCollection currentTraitCollection].userInterfaceStyle == UIUserInterfaceStyleDark)
    {
      self.window.rootViewController.view.backgroundColor = [[UIColor alloc] initWithRed:0.0f green:0.0f blue:0.0f alpha:1];
    } else {
      self.window.rootViewController.view.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];
    }
  } else {
    self.window.rootViewController.view.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];
  }
}

@end

File: ios/AtBRootView.swift
import React

@objc(AtBRootView)
class AtBRootView: RCTRootView {

    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
        super.traitCollectionDidChange(previousTraitCollection)
     
        print("traitCollectionDidChange");
        print(traitCollection.displayScale);
        print(traitCollection.userInterfaceStyle);
      
        if #available(iOS 13.0, *) {
            if traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle {
                setBackgroundByTrait()
            }
        }
    }

    func setBackgroundByTrait() {
        if #available(iOS 13.0, *) {
            if traitCollection.userInterfaceStyle == .dark {
                backgroundColor = UIColor(red: 0.0, green: 0.0, blue: 0.0, alpha: 1.0)
            } else {
                backgroundColor = UIColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 1.0)
            }
        } else {
            backgroundColor = UIColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 1.0)
        }
    }
}
